extension/src/ALObject/XliffIdToken.ts

```diff
@@ -17,7 +17,9 @@ export class XliffIdToken {
   }
   public set name(v: string) {
     if (v.startsWith('"') && v.endsWith('"')) {
-      v = v.substr(1, v.length - 2);
+      if (!v.substr(1, v.length - 2).includes('"')) {
+        v = v.substr(1, v.length - 2);
```

Comment:
`substr` is depreacated. Would be nice to replace with `slice`, if possible in this PR. There is a slight difference, the second parameter of `substr` is length. Whereas `slice` has index. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/substr

Reply:
This kind of change seems dangerous. So easy to screw things up when changing the second parameter...

So I do not want to change this right now, but going forward I will start to use `slice` instead of `substr`. And we'll do a slow migration over to `slice` when we are touching code and can do a thorough testing of the change.

Ok?

Reply:
Sounds good. There's (probably) no rush and that migration won't pass without mistakes :sweat_smile: